### ISSUE TITLE
Audio delays and performance issues in Safari on iOS/iPadOS

### ISSUE THREAD

**Issue by morgant** [wwwtyro/keyzen#18]
When used in Safari on iPadOS (14.x) with the Apple Magic Keyboard, the audio playback stutters and causes general performance issues.

Looking at `keyzen.js`, this may be due to the fact that the `Audio` element is created, loaded, and then played every time `keyHandler()` or `level_up()` is called (see https://github.com/wwwtyro/keyzen/blob/9a756471984c24b22300d5a3332f5cbdcc9d05d2/keyzen.js#L95, https://github.com/wwwtyro/keyzen/blob/9a756471984c24b22300d5a3332f5cbdcc9d05d2/keyzen.js#L101, and https://github.com/wwwtyro/keyzen/blob/9a756471984c24b22300d5a3332f5cbdcc9d05d2/keyzen.js#L132). I believe that Safari on iOS/iPadOS doesn't cache media files, so it may be re-fetching the `.wav`s on every instantiation of `Audio`.

I'd suggest starting by instantiating the three `Audio` elements only once and re-using them. If that doesn't work, migrating to the [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) would probably fix it as they all could be loaded under one `AudioContext`.

**Comment by morgant** [comment]
Looks like PR #2 from way back in the day would have addressed this, but was closed. I'll clean up and submit a new PR.

**Comment by morgant** [comment]
I did a quick test of instantiating the sounds once, but upon testing I realized why it wasn't implemented that way: when typing quickly, the same sound will not be played for every key press as it may still/already be playing. It makes it sound/feel like it's not properly registering key presses.

So, I think that migrating to the Web Audio API is probably the best bet.

**Comment by morgant** [comment]
I have created an implementation using the [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) and have tested in Chrome on desktop and Safari on iPadOS (14.x) and it completely resolves the stuttering audio & performance issues.

There's some minor fallout, including:

* Requires newer browser versions (though not super recent, see https://caniuse.com/mdn-api_response_arraybuffer & https://caniuse.com/mdn-api_baseaudiocontext_decodeaudiodata_returns_promise).
* Can't use [jQuery.ajax()](https://api.jquery.com/jquery.ajax/) because it doesn't support the 'arraybuffer' data type.
* Audio won't work if `index.html` is loaded locally via `file:///` due to CORS as the `.wav` files must be loaded from a web server via [XHR](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)/`fetch()`. Unfortunately, the Web Audio API can't reasonably [copy data from an Audio media element to an AudioBuffer](https://github.com/WebAudio/web-audio-api/issues/1872).

It would be very easy to fall back to the Audio media element playback implementation if the newer browser support is an issue for anyone.

I'll create a PR for review before I merge. Feedback is definitely welcome.

**Comment by morgant** [comment]
PR #19 has been merged, so this should be all set.
